server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -294,6 +296,42 @@ public Iterator<Setting<?>> settings() {
         Property.Final
     );
 
+    /**
+     * Used to specify if the index translog operations should be persisted in the remote store.
+     */
+    public static final Setting<Boolean> INDEX_REMOTE_TRANSLOG_STORE_ENABLED_SETTING = Setting.boolSetting(
+        SETTING_REMOTE_TRANSLOG_STORE_ENABLED,
+        false,
+        new Setting.Validator<>() {
+
+            @Override
+            public void validate(final Boolean value) {}
```

Comment:
Isn't this called when Settings is `null`? Should we throw an exception here since we can't be sure if remote store is even configured?

Reply:
I guess `Setting#boolSettings` has an implicit parser

https://github.com/opensearch-project/OpenSearch/blob/3aef125d0dbf2f29d9bf00b84cb52892a27a5328/server/src/main/java/org/opensearch/common/settings/Setting.java#L1700-L1702

which gets invoked on every prior the validate call anyways. 

https://github.com/opensearch-project/OpenSearch/blob/3aef125d0dbf2f29d9bf00b84cb52892a27a5328/server/src/main/java/org/opensearch/common/settings/Setting.java#L468-L495

Let me know if you still want an explicit check?